src/app.js

```diff
@@ -31,11 +32,14 @@ io.of('/game').on('connection', (socket) => {
   socket.on(MOVE_PLAYED, async ({ gameId, playerId, move }) => {
     const game = await dynamo.getGameById(gameId)
     const { piece, position } = move
-    game.board[position] = piece
-    const [otherPlayer] = game.players.filter((player) => player.id !== playerId)
-    game.playerTurn = otherPlayer.id
-    await dynamo.updateGame(game)
-    io.of('/game').to(gameId).emit(GAME_STATE, game)
+
+    if (isValidMove(game, playerId, position)) {
+      game.board[position] = piece
+      const [otherPlayer] = game.players.filter((player) => player.id !== playerId)
+      game.playerTurn = otherPlayer.id
+      await dynamo.updateGame(game)
+      io.of('/game').to(gameId).emit(GAME_STATE, game)
+    }
```

Comment:
If we invert the check and call `isMoveInvalid()`, we would alleviate code nesting. Something like:
```js
if(isMoveInvalid(game,playerId,position))
   return
// and then the logic 👇 ....
```